dsn_test.go

```diff
@@ -1233,6 +1233,31 @@ func TestDSN(t *testing.T) {
 			},
 			dsn: "u:p@account-name.cn-region.snowflakecomputing.cn:443?account=account-name&ocspFailOpen=true&region=cn-region&validateDefaultParameters=true",
 		},
+		{
+			cfg: &Config{
+				User:     "u",
+				Password: "p",
+				Account:  "account.us-west-2",
+			},
+			dsn: "u:p@account.snowflakecomputing.com:443?ocspFailOpen=true&validateDefaultParameters=true",
+		},
+		{
+			cfg: &Config{
+				User:     "u",
+				Password: "p",
+				Account:  "account_us-west-2",
+			},
+			dsn: "u:p@account_us-west-2.snowflakecomputing.com:443?ocspFailOpen=true&validateDefaultParameters=true",
+		},
+		{
+			cfg: &Config{
+				User:     "u",
+				Password: "p",
+				Account:  "account",
+				Host:     "account.us-west-2.snowflakecomputing.com",
```

Comment:
we support also `account.us-west-2.aws` - there also dsn should be shortened?

Reply:
then the next question is that we should correct every notation in every driver, which i agree is the ideal final goal

Reply:
added `SNOW-1858448` to track the more comprehensive effort